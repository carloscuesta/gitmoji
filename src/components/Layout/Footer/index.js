// @flow
import React, { type Element } from 'react'
import Link from 'next/link'

import Icon from '../../Icon'

const Footer = (): Element<'footer'> => (
  <footer className="footer">
    <div className="wrap">
      <div className="row middle-xs">
        <div className="col-sm-6 made-with-love">
          <h3>
            Made with <Icon name="heart" /> by{' '}
            <a href="https://carloscuesta.me">Carlos Cuesta</a>
          </h3>
        </div>
        <div className="col-sm-6 footer-nav">
          <nav>
            <Link href="/about">
              <a>About</a>
            </Link>
            <Link href="/contributors">
              <a>Contributors</a>
            </Link>
            <a href="https://github.com/carloscuesta/gitmoji">GitHub</a>
          </nav>
        </div>
      </div>
    </div>
  </footer>
)

export default Footer
