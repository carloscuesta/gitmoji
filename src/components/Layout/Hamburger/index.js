// @flow
import React, { type Element } from 'react'
import Router from 'next/router'

import MenuLink from './MenuLink'
import OpenIcon from './OpenIcon'
import CloseIcon from './CloseIcon'
import styles from './styles.module.css'

const Hamburger = (): Element<'div'> => {
  const [isOpen, setIsOpen] = React.useState(false)

  React.useEffect(() => {
    const onRouteChangeStart = () => {
      setIsOpen(false)
    }

    Router.events.on('routeChangeStart', onRouteChangeStart)

    return () => Router.events.off('routeChangeStart', onRouteChangeStart)
  }, [])

  return (
    <div className={styles.hamburger}>
      <button
        aria-label="Open navigation menu"
        className={styles.button}
        onClick={() => setIsOpen(true)}
      >
        <OpenIcon />
      </button>

      {isOpen && (
        <nav className={styles.menu}>
          <div className={styles.closeContainer}>
            <button
              aria-label="Close navigation menu"
              className={styles.button}
              onClick={() => setIsOpen(false)}
            >
              <CloseIcon />
            </button>
          </div>

          <ul className={styles.links}>
            <li>
              <MenuLink href="/" text="Home" />
            </li>
            <li>
              <MenuLink href="/about" text="About" />
            </li>
            <li>
              <MenuLink href="/contributors" text="Contributors" />
            </li>
            <li>
              <MenuLink
                href="https://github.com/carloscuesta/gitmoji"
                text="GitHub"
              />
            </li>
          </ul>
        </nav>
      )}
    </div>
  )
}

export default Hamburger
