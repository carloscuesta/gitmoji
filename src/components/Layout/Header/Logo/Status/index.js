// @flow
import React, { type Node } from 'react'

import Joy from './Joy'
import Loved from './Loved'
import Sexy from './Sexy'
import Smiling from './Smiling'
import Sunglasses from './Sunglasses'
import Tongue from './Tongue'

export const LOGO_STATUSES: Object = {
  JOY: 'joy',
  LOVED: 'loved',
  SEXY: 'sexy',
  SMILING: 'smiling',
  SUNGLASSES: 'sunglasses',
  TONGUE: 'tongue',
}

type Props = { status: $Keys<typeof LOGO_STATUSES> }

const Status = (props: Props): Node => {
  switch (props.status) {
    case LOGO_STATUSES.JOY:
      return <Joy />
    case LOGO_STATUSES.LOVED:
      return <Loved />
    case LOGO_STATUSES.SEXY:
      return <Sexy />
    case LOGO_STATUSES.SMILING:
      return <Smiling />
    case LOGO_STATUSES.SUNGLASSES:
      return <Sunglasses />
    case LOGO_STATUSES.TONGUE:
      return <Tongue />
  }

  return null
}

export default Status
