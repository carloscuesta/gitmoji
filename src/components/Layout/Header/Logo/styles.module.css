.logo {
  width: 100%;
  height: 115px;
}
