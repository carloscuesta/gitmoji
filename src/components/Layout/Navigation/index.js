// @flow
import React, { type Element } from 'react'
import Link from 'next/link'

const Navigation = (): Element<'nav'> => (
  <nav className="nav row middle-xs">
    <input className="c-hamburger__input" id="hamburger-menu" type="checkbox" />
    <label className="c-hamburger c-hamburger--htx" htmlFor="hamburger-menu">
      <span />
    </label>
    <div className="modal">
      <div className="modal-content">
        <ul className="navigation">
          <li className="item">
            <Link href="/">
              <a
                onMouseDown={() =>
                  window.ga('send', 'event', 'Menu', 'Gitmoji')
                }
              >
                Home
              </a>
            </Link>
          </li>
          <li className="item">
            <Link href="/about">
              <a
                onMouseDown={() => window.ga('send', 'event', 'Menu', 'About')}
              >
                About
              </a>
            </Link>
          </li>
          <li className="item">
            <Link href="/contributors">
              <a
                onMouseDown={() =>
                  window.ga('send', 'event', 'Menu', 'Contributors')
                }
              >
                Contributors
              </a>
            </Link>
          </li>
          <li className="item">
            <a
              href="https://github.com/carloscuesta/gitmoji"
              onMouseDown={() => window.ga('send', 'event', 'Menu', 'GitHub')}
            >
              GitHub
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
)

export default Navigation
