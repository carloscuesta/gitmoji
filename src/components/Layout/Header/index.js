// @flow
import React, { type Element } from 'react'

import Button from '../../Button'
import Logo from '../Logo'

type Props = { withHeadline: boolean, withSocialButtons: boolean }

const Header = (props: Props): Element<'header'> => (
  <header className="header">
    <Logo />
    {props.withHeadline && <h2>An emoji guide for your commit messages</h2>}
    {props.withSocialButtons && (
      <div className="header-buttons">
        <Button
          icon="star"
          link="https://github.com/carloscuesta/gitmoji"
          text="GitHub"
        />
        <Button
          icon="twitter"
          link={
            'https://twitter.com/intent/tweet?text=gitmoji' +
            '%20%E2%80%93%20An%20%23emoji%20guide%20for%20your%20commit' +
            '%20messages%20by%20%40crloscuesta%20%F0%9F%98%8D%F0%9F%98%9C' +
            '&url=https://gitmoji.carloscuesta.me'
          }
          target="_blank"
          text="Tweet"
        />
      </div>
    )}
  </header>
)

export default Header
