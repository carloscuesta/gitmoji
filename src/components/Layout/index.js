// @flow
import * as React from 'react'

import { IconDefinitions } from '../Icon'
import Header from './Header'
import Navigation from './Navigation'
import Footer from './Footer'

type Props = { headerWithSocialButtons: boolean, children: React.Node }

const Layout = (props: Props): React.Node => (
  <>
    <IconDefinitions />
    <Navigation />
    <Header withHeadline withSocialButtons={props.headerWithSocialButtons} />
    <main className="wrap">{props.children}</main>
    <Footer />
  </>
)

export default Layout
