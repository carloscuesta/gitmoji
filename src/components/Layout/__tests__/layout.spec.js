import renderer from 'react-test-renderer'

import Layout from '../index'
import { getIconByStatus } from '../Logo'
import LOGO_STATUSES from '../Logo/statuses'
import * as stubs from './stubs'

describe('Layout', () => {
  beforeAll(() => {
    Math.random = jest.fn().mockReturnValue(1)
  })

  it('should render the component with social buttons', () => {
    const wrapper = renderer.create(
      <Layout {...stubs.props}>
        <p>Some children</p>
      </Layout>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the component without social buttons', () => {
    const wrapper = renderer.create(
      <Layout>
        <p>Some children</p>
      </Layout>
    )
    expect(wrapper).toMatchSnapshot()
  })

  describe('Logo', () => {
    describe('getIconByStatus', () => {
      const statuses = Object.values(LOGO_STATUSES).map((status) => status)

      statuses.forEach((status) => {
        it('should render component for status ' + status, () => {
          expect(getIconByStatus(status)).toMatchSnapshot()
        })
      })
    })
  })
})
