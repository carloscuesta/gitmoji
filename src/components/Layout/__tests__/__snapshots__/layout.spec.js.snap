// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Logo getIconByStatus should render component for status joy 1`] = `<Joy />`;

exports[`Layout Logo getIconByStatus should render component for status loved 1`] = `<Loved />`;

exports[`Layout Logo getIconByStatus should render component for status sexy 1`] = `<Sexy />`;

exports[`Layout Logo getIconByStatus should render component for status smiling 1`] = `<Smiling />`;

exports[`Layout Logo getIconByStatus should render component for status sunglasses 1`] = `<Sunglasses />`;

exports[`Layout Logo getIconByStatus should render component for status tongue 1`] = `<Tongue />`;

exports[`Layout should render the component with social buttons 1`] = `
Array [
  <svg
    height={0}
    style={
      Object {
        "height": 0,
        "position": "absolute",
        "width": 0,
      }
    }
    version="1.1"
    width={0}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <symbol
        id="icon-heart"
        viewBox="0 0 64 64"
      >
        <title>
          heart
        </title>
        <path
          className="heart"
          d="m61.1 18.2c-6.4-17-27.2-9.4-29.1-.9-2.6-9-22.9-15.7-29.1.9-6.9 18.5 26.7 35.1 29.1 37.8 2.4-2.2 36-19.6 29.1-37.8"
          fill="#ff5a79"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 64 64"
      >
        <title>
          star
        </title>
        <path
          className="twitter"
          d="M62,25.2H39.1L32,3l-7.1,22.2H2l18.5,13.7l-7,22.1L32,47.3L50.5,61l-7.1-22.2L62,25.2z"
          fill="#FFDD67"
        />
      </symbol>
      <symbol
        id="icon-twitter"
        viewBox="0 0 64 64"
      >
        <title>
          twitter
        </title>
        <g
          fill="#42ade2"
        >
          <path
            d="m59.8 24.3c0 0 1.1-6.2-3.5-3.4 0 0-.4-6.3-4.3-1.9 0 0-2.1-3.9-4.4-.3-3.1 4.8-5.2 12.4-3.2 25l3.8-2.5c2.7-7.9 12.4-8.8 13.7-13.1.9-3-2.1-3.8-2.1-3.8"
          />
          <path
            d="m22.1 17.6l-9.9 3.6c2.2-12 16.6-11.2 16.6-11.2s-6.8 3.2-6.7 7.6"
          />
          <path
            d="m23.7 19.8l-10.5 1.4c4.8-11.2 18.7-7.3 18.7-7.3s-7.3 1.6-8.2 5.9"
          />
        </g>
        <g
          fill="#ffd93b"
        >
          <path
            d="m2 29l5.4-1.4v3.6c0-.1-3.3-.6-5.4-2.2"
          />
          <path
            d="M7.4,27.5L2,24.8c3.6-2.8,7.7-1.9,7.7-1.9L7.4,27.5z"
          />
        </g>
        <g
          fill="#e08828"
        >
          <path
            d="m33.8 53h-2.1v7.9c-.3.1-2.1-.1-2.9-.1-1.8 0-3.3 1.3-3.3 1.3h8.3v-9.1"
          />
          <path
            d="m25 53h-2.1v7.9c-.3.1-2.1-.1-2.9-.1-1.8 0-3.3 1.3-3.3 1.3h8.3v-9.1"
          />
          <path
            d="m54 36.2c3.9 0-4.1 17.5-23.3 17.5-13 0-23.9-5.2-23.9-21.5 0-10.1 6.4-18.3 19.5-15 13.3 3.5 6.5 19 27.7 19"
            fill="#42ade2"
          />
          <path
            d="m37.6 51.7c-15.6 0-14-12-27.9-11.2 5.1 15.8 27.9 11.2 27.9 11.2"
            fill="#fff"
          />
          <path
            d="m39.1 29.2c-10-9.8-20.2 6.2-7.9 12.6 12.1 6.2 20.4-4.8 20.4-4.8s-6.1-1.5-12.5-7.8"
            fill="#297b9d"
          />
        </g>
        <circle
          cx="15.1"
          cy="24.9"
          fill="#3e4347"
          r="2.5"
        />
      </symbol>
    </defs>
  </svg>,
  <nav
    className="nav row middle-xs"
  >
    <input
      className="c-hamburger__input"
      id="hamburger-menu"
      type="checkbox"
    />
    <label
      className="c-hamburger c-hamburger--htx"
      htmlFor="hamburger-menu"
    >
      <span />
    </label>
    <div
      className="modal"
    >
      <div
        className="modal-content"
      >
        <ul
          className="navigation"
        >
          <li
            className="item"
          >
            <a
              href="/"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="item"
          >
            <a
              href="/about"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="item"
          >
            <a
              href="/contributors"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
            >
              Contributors
            </a>
          </li>
          <li
            className="item"
          >
            <a
              href="https://github.com/carloscuesta/gitmoji"
              onMouseDown={[Function]}
            >
              GitHub
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <header
    className="header"
  >
    <svg
      className="gitmoji-logo"
      height="137px"
      version="1.1"
      viewBox="0 0 457 137"
      width="457px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <title>
        gitmoji
      </title>
      <g
        fill="none"
        fillRule="evenodd"
        id="gitmoji"
      >
        <g
          id="Logo"
          transform="translate(-270 -430)"
        >
          <g
            id="gitmoji"
            transform="translate(270 430)"
          >
            <path
              d="m17.5 106c0.6 4 2.4 7 5.3 10 2.9 2 6.6 4 11.2 4 6.1 0 10.7-2 14-5s4.9-8 4.9-15.1v-5.1c-1.5 2.3-3.8 4.4-7.1 6.2-3.2 2-7.5 3-12.6 3-4.7 0-9.1-1-13.1-3-4.1-1.5-7.6-3.9-10.5-7-2.91-3-5.22-6.7-6.9-10.8-1.68-4.2-2.52-8.8-2.52-13.8 0.004-4.7 0.82-9.1 2.45-13.3s3.89-7.9 6.77-11c2.9-3.2 6.3-5.7 10.4-7.5 4-1.8 8.5-2.7 13.4-2.7 5.6 0 10.1 1 13.4 3 3.2 2 5.5 4.2 6.7 6.6v-8.3h18.5v63.2c0 4.6-0.7 9.6-2.1 13.6-1.4 5-3.6 9-6.6 12-3 4-6.8 6-11.5 8s-10.4 4-17 4c-4.9 0-9.3-1-13.3-3-4-1-7.5-3-10.5-5-2.94-3-5.34-5-7.16-8-1.82-4-2.98-7-3.46-10l17.3-5zm19-18.6c4.9 0 8.9-1.6 12-4.9 3.1-3.2 4.7-7.6 4.7-13.1s-1.7-9.8-4.9-13c-3.3-3.2-7.2-4.8-11.8-4.8-2.4 0-4.7 0.4-6.7 1.2-2.1 0.8-3.9 2-5.4 3.6-1.6 1.6-2.8 3.5-3.6 5.7-0.9 2.2-1.3 4.6-1.3 7.3 0 5.6 1.6 10 4.7 13.2 3.2 3.2 7.3 4.8 12.3 4.8zm54.7 19.6v-70.8h18.8v70.8h-18.8zm-2.3-94c0-3.39 1.1-6.22 3.4-8.53 2.3-2.3 5.1-3.45 8.7-3.45 3 0 6 1.15 8 3.45 2 2.31 4 5.14 4 8.53 0 3.1-2 5.9-4 8.2s-5 3.4-8 3.4c-3.6 0-6.4-1.1-8.7-3.4s-3.4-5.1-3.4-8.2zm65.1 23.2h15v16.9h-15v29.7c0 3.1 1 5.2 2 6.4 2 1.2 4 1.8 7 1.8 1 0 2 0 3-0.1s2-0.3 3-0.5v15.6c-1 1-3 1-4 1-2 1-4 1-7 1-7 0-13-2-17-6-4-3.6-6-9-6-15.9v-33h-12v-16.9h3c4 0 7-1.1 8-3.3 2-2.1 3-4.8 3-8v-9.9h17v21.2zm30 70.8v-70.8h18v8.6c1-1.7 2-3.3 4-4.6 1-1.4 3-2.5 5-3.3 2-0.9 4-1.6 6-2 2-0.5 4-0.8 6-0.8 5 0 9 1 13 3.1 3 2 6 4.9 8 8.7 3-4.3 6-7.3 10-9.1s8-2.7 12-2.7 7 0.5 10 1.5c3 1.1 6 2.7 8 4.9 3 2.2 4 5 6 8.4 1 3.4 2 7.5 2 12.2v45.9h-19v-42c0-3.9 0-7.1-2-9.6-2-2.6-6-3.8-10-3.8s-7 1.3-10 4.1c-2 2.7-3 6-3 9.8v41.5h-19v-42c0-3.9-1-7.1-3-9.6-2-2.6-5-3.8-10-3.8-4 0-7 1.3-9 4-3 2.7-4 6-4 9.9v41.5h-19zm159-15.3c3 0 5-0.4 7-1.3 2-0.8 4-2.1 6-3.8 1-1.7 3-3.7 4-6.2s1-5.5 1-8.8c0-3.4 0-6.3-1-8.8s-3-4.6-4-6.3c-2-1.7-4-2.9-6-3.8s-4-1.3-7-1.3c-2 0-4 0.4-6 1.3-3 0.9-5 2.1-6 3.8-2 1.7-3 3.8-4 6.3s-2 5.4-2 8.8c0 3.3 1 6.3 2 8.8s2 4.5 4 6.2c1 1.7 3 3 6 3.8 2 0.9 4 1.3 6 1.3zm0-57.7c6 0 11 0.9 15 2.8 5 1.9 9 4.5 12 7.8s6 7.2 8 11.9c2 4.6 2 9.6 2 15.1 0 5.4 0 10.5-2 15-2 4.6-5 8.5-8 11.9-3 3.5-7 5.5-12 7.5-4 2-9 3-15 3-5 0-10-1-14-3-5-2-9-4-12-7.5-4-3.4-6-7.3-8-11.9-2-4.5-3-9.6-3-15 0-5.5 1-10.5 3-15.1 2-4.7 4-8.6 8-11.9 3-3.3 7-5.9 12-7.8 4-1.9 9-2.8 14-2.8zm53 2.2h19v76.8c0 4-1 7-2 10-1 2-2 5-4 7s-4 4-7 5-6 2-9 2-6-1-8-1c-2-1-3-1-3-1v-16h2c1 1 2 1 4 1 3 0 5-1 6-3 1-1 2-3 2-6v-74.8zm-3-23.4c0-3.34 1-6.17 4-8.47 2-2.31 5-3.46 8-3.46 3 0.002 6 1.15 9 3.46 2 2.3 3 5.13 3 8.47 0 3.3-1 6-3 8.3-3 2.3-6 3.4-9 3.4s-6-1.1-8-3.4c-3-2.3-4-5-4-8.3zm42 94.2v-70.8h19v70.8h-19zm-2-94c0-3.39 1-6.22 3-8.53 3-2.3 5-3.45 9-3.45 3 0 6 1.15 8 3.45 2 2.31 4 5.14 4 8.53 0 3.1-2 5.9-4 8.2s-5 3.4-8 3.4c-4 0-6-1.1-9-3.4-2-2.3-3-5.1-3-8.2z"
              fill="#000"
            />
          </g>
        </g>
      </g>
    </svg>
    <h2>
      An emoji guide for your commit messages
    </h2>
    <div
      className="header-buttons"
    >
      <a
        className="btn btn-pink"
        href="https://github.com/carloscuesta/gitmoji"
      >
        <svg
          className="icon icon-star"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
        GitHub
      </a>
      <a
        className="btn btn-pink"
        href="https://twitter.com/intent/tweet?text=gitmoji%20%E2%80%93%20An%20%23emoji%20guide%20for%20your%20commit%20messages%20by%20%40crloscuesta%20%F0%9F%98%8D%F0%9F%98%9C&url=https://gitmoji.carloscuesta.me"
        target="_blank"
      >
        <svg
          className="icon icon-twitter"
        >
          <use
            xlinkHref="#icon-twitter"
          />
        </svg>
        Tweet
      </a>
    </div>
  </header>,
  <main
    className="wrap"
  >
    <p>
      Some children
    </p>
  </main>,
  <footer
    className="footer"
  >
    <div
      className="wrap"
    >
      <div
        className="row middle-xs"
      >
        <div
          className="col-sm-6 made-with-love"
        >
          <h3>
            Made with 
            <svg
              className="icon icon-heart"
            >
              <use
                xlinkHref="#icon-heart"
              />
            </svg>
             by
             
            <a
              href="https://carloscuesta.me"
            >
              Carlos Cuesta
            </a>
          </h3>
        </div>
        <div
          className="col-sm-6 footer-nav"
        >
          <nav>
            <a
              href="/about"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              About
            </a>
            <a
              href="/contributors"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Contributors
            </a>
            <a
              href="https://github.com/carloscuesta/gitmoji"
            >
              GitHub
            </a>
          </nav>
        </div>
      </div>
    </div>
  </footer>,
]
`;

exports[`Layout should render the component without social buttons 1`] = `
Array [
  <svg
    height={0}
    style={
      Object {
        "height": 0,
        "position": "absolute",
        "width": 0,
      }
    }
    version="1.1"
    width={0}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <symbol
        id="icon-heart"
        viewBox="0 0 64 64"
      >
        <title>
          heart
        </title>
        <path
          className="heart"
          d="m61.1 18.2c-6.4-17-27.2-9.4-29.1-.9-2.6-9-22.9-15.7-29.1.9-6.9 18.5 26.7 35.1 29.1 37.8 2.4-2.2 36-19.6 29.1-37.8"
          fill="#ff5a79"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 64 64"
      >
        <title>
          star
        </title>
        <path
          className="twitter"
          d="M62,25.2H39.1L32,3l-7.1,22.2H2l18.5,13.7l-7,22.1L32,47.3L50.5,61l-7.1-22.2L62,25.2z"
          fill="#FFDD67"
        />
      </symbol>
      <symbol
        id="icon-twitter"
        viewBox="0 0 64 64"
      >
        <title>
          twitter
        </title>
        <g
          fill="#42ade2"
        >
          <path
            d="m59.8 24.3c0 0 1.1-6.2-3.5-3.4 0 0-.4-6.3-4.3-1.9 0 0-2.1-3.9-4.4-.3-3.1 4.8-5.2 12.4-3.2 25l3.8-2.5c2.7-7.9 12.4-8.8 13.7-13.1.9-3-2.1-3.8-2.1-3.8"
          />
          <path
            d="m22.1 17.6l-9.9 3.6c2.2-12 16.6-11.2 16.6-11.2s-6.8 3.2-6.7 7.6"
          />
          <path
            d="m23.7 19.8l-10.5 1.4c4.8-11.2 18.7-7.3 18.7-7.3s-7.3 1.6-8.2 5.9"
          />
        </g>
        <g
          fill="#ffd93b"
        >
          <path
            d="m2 29l5.4-1.4v3.6c0-.1-3.3-.6-5.4-2.2"
          />
          <path
            d="M7.4,27.5L2,24.8c3.6-2.8,7.7-1.9,7.7-1.9L7.4,27.5z"
          />
        </g>
        <g
          fill="#e08828"
        >
          <path
            d="m33.8 53h-2.1v7.9c-.3.1-2.1-.1-2.9-.1-1.8 0-3.3 1.3-3.3 1.3h8.3v-9.1"
          />
          <path
            d="m25 53h-2.1v7.9c-.3.1-2.1-.1-2.9-.1-1.8 0-3.3 1.3-3.3 1.3h8.3v-9.1"
          />
          <path
            d="m54 36.2c3.9 0-4.1 17.5-23.3 17.5-13 0-23.9-5.2-23.9-21.5 0-10.1 6.4-18.3 19.5-15 13.3 3.5 6.5 19 27.7 19"
            fill="#42ade2"
          />
          <path
            d="m37.6 51.7c-15.6 0-14-12-27.9-11.2 5.1 15.8 27.9 11.2 27.9 11.2"
            fill="#fff"
          />
          <path
            d="m39.1 29.2c-10-9.8-20.2 6.2-7.9 12.6 12.1 6.2 20.4-4.8 20.4-4.8s-6.1-1.5-12.5-7.8"
            fill="#297b9d"
          />
        </g>
        <circle
          cx="15.1"
          cy="24.9"
          fill="#3e4347"
          r="2.5"
        />
      </symbol>
    </defs>
  </svg>,
  <nav
    className="nav row middle-xs"
  >
    <input
      className="c-hamburger__input"
      id="hamburger-menu"
      type="checkbox"
    />
    <label
      className="c-hamburger c-hamburger--htx"
      htmlFor="hamburger-menu"
    >
      <span />
    </label>
    <div
      className="modal"
    >
      <div
        className="modal-content"
      >
        <ul
          className="navigation"
        >
          <li
            className="item"
          >
            <a
              href="/"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="item"
          >
            <a
              href="/about"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="item"
          >
            <a
              href="/contributors"
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
            >
              Contributors
            </a>
          </li>
          <li
            className="item"
          >
            <a
              href="https://github.com/carloscuesta/gitmoji"
              onMouseDown={[Function]}
            >
              GitHub
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <header
    className="header"
  >
    <svg
      className="gitmoji-logo"
      height="137px"
      version="1.1"
      viewBox="0 0 457 137"
      width="457px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <title>
        gitmoji
      </title>
      <g
        fill="none"
        fillRule="evenodd"
        id="gitmoji"
      >
        <g
          id="Logo"
          transform="translate(-270 -430)"
        >
          <g
            id="gitmoji"
            transform="translate(270 430)"
          >
            <path
              d="m17.5 106c0.6 4 2.4 7 5.3 10 2.9 2 6.6 4 11.2 4 6.1 0 10.7-2 14-5s4.9-8 4.9-15.1v-5.1c-1.5 2.3-3.8 4.4-7.1 6.2-3.2 2-7.5 3-12.6 3-4.7 0-9.1-1-13.1-3-4.1-1.5-7.6-3.9-10.5-7-2.91-3-5.22-6.7-6.9-10.8-1.68-4.2-2.52-8.8-2.52-13.8 0.004-4.7 0.82-9.1 2.45-13.3s3.89-7.9 6.77-11c2.9-3.2 6.3-5.7 10.4-7.5 4-1.8 8.5-2.7 13.4-2.7 5.6 0 10.1 1 13.4 3 3.2 2 5.5 4.2 6.7 6.6v-8.3h18.5v63.2c0 4.6-0.7 9.6-2.1 13.6-1.4 5-3.6 9-6.6 12-3 4-6.8 6-11.5 8s-10.4 4-17 4c-4.9 0-9.3-1-13.3-3-4-1-7.5-3-10.5-5-2.94-3-5.34-5-7.16-8-1.82-4-2.98-7-3.46-10l17.3-5zm19-18.6c4.9 0 8.9-1.6 12-4.9 3.1-3.2 4.7-7.6 4.7-13.1s-1.7-9.8-4.9-13c-3.3-3.2-7.2-4.8-11.8-4.8-2.4 0-4.7 0.4-6.7 1.2-2.1 0.8-3.9 2-5.4 3.6-1.6 1.6-2.8 3.5-3.6 5.7-0.9 2.2-1.3 4.6-1.3 7.3 0 5.6 1.6 10 4.7 13.2 3.2 3.2 7.3 4.8 12.3 4.8zm54.7 19.6v-70.8h18.8v70.8h-18.8zm-2.3-94c0-3.39 1.1-6.22 3.4-8.53 2.3-2.3 5.1-3.45 8.7-3.45 3 0 6 1.15 8 3.45 2 2.31 4 5.14 4 8.53 0 3.1-2 5.9-4 8.2s-5 3.4-8 3.4c-3.6 0-6.4-1.1-8.7-3.4s-3.4-5.1-3.4-8.2zm65.1 23.2h15v16.9h-15v29.7c0 3.1 1 5.2 2 6.4 2 1.2 4 1.8 7 1.8 1 0 2 0 3-0.1s2-0.3 3-0.5v15.6c-1 1-3 1-4 1-2 1-4 1-7 1-7 0-13-2-17-6-4-3.6-6-9-6-15.9v-33h-12v-16.9h3c4 0 7-1.1 8-3.3 2-2.1 3-4.8 3-8v-9.9h17v21.2zm30 70.8v-70.8h18v8.6c1-1.7 2-3.3 4-4.6 1-1.4 3-2.5 5-3.3 2-0.9 4-1.6 6-2 2-0.5 4-0.8 6-0.8 5 0 9 1 13 3.1 3 2 6 4.9 8 8.7 3-4.3 6-7.3 10-9.1s8-2.7 12-2.7 7 0.5 10 1.5c3 1.1 6 2.7 8 4.9 3 2.2 4 5 6 8.4 1 3.4 2 7.5 2 12.2v45.9h-19v-42c0-3.9 0-7.1-2-9.6-2-2.6-6-3.8-10-3.8s-7 1.3-10 4.1c-2 2.7-3 6-3 9.8v41.5h-19v-42c0-3.9-1-7.1-3-9.6-2-2.6-5-3.8-10-3.8-4 0-7 1.3-9 4-3 2.7-4 6-4 9.9v41.5h-19zm159-15.3c3 0 5-0.4 7-1.3 2-0.8 4-2.1 6-3.8 1-1.7 3-3.7 4-6.2s1-5.5 1-8.8c0-3.4 0-6.3-1-8.8s-3-4.6-4-6.3c-2-1.7-4-2.9-6-3.8s-4-1.3-7-1.3c-2 0-4 0.4-6 1.3-3 0.9-5 2.1-6 3.8-2 1.7-3 3.8-4 6.3s-2 5.4-2 8.8c0 3.3 1 6.3 2 8.8s2 4.5 4 6.2c1 1.7 3 3 6 3.8 2 0.9 4 1.3 6 1.3zm0-57.7c6 0 11 0.9 15 2.8 5 1.9 9 4.5 12 7.8s6 7.2 8 11.9c2 4.6 2 9.6 2 15.1 0 5.4 0 10.5-2 15-2 4.6-5 8.5-8 11.9-3 3.5-7 5.5-12 7.5-4 2-9 3-15 3-5 0-10-1-14-3-5-2-9-4-12-7.5-4-3.4-6-7.3-8-11.9-2-4.5-3-9.6-3-15 0-5.5 1-10.5 3-15.1 2-4.7 4-8.6 8-11.9 3-3.3 7-5.9 12-7.8 4-1.9 9-2.8 14-2.8zm53 2.2h19v76.8c0 4-1 7-2 10-1 2-2 5-4 7s-4 4-7 5-6 2-9 2-6-1-8-1c-2-1-3-1-3-1v-16h2c1 1 2 1 4 1 3 0 5-1 6-3 1-1 2-3 2-6v-74.8zm-3-23.4c0-3.34 1-6.17 4-8.47 2-2.31 5-3.46 8-3.46 3 0.002 6 1.15 9 3.46 2 2.3 3 5.13 3 8.47 0 3.3-1 6-3 8.3-3 2.3-6 3.4-9 3.4s-6-1.1-8-3.4c-3-2.3-4-5-4-8.3zm42 94.2v-70.8h19v70.8h-19zm-2-94c0-3.39 1-6.22 3-8.53 3-2.3 5-3.45 9-3.45 3 0 6 1.15 8 3.45 2 2.31 4 5.14 4 8.53 0 3.1-2 5.9-4 8.2s-5 3.4-8 3.4c-4 0-6-1.1-9-3.4-2-2.3-3-5.1-3-8.2z"
              fill="#000"
            />
          </g>
        </g>
      </g>
    </svg>
    <h2>
      An emoji guide for your commit messages
    </h2>
  </header>,
  <main
    className="wrap"
  >
    <p>
      Some children
    </p>
  </main>,
  <footer
    className="footer"
  >
    <div
      className="wrap"
    >
      <div
        className="row middle-xs"
      >
        <div
          className="col-sm-6 made-with-love"
        >
          <h3>
            Made with 
            <svg
              className="icon icon-heart"
            >
              <use
                xlinkHref="#icon-heart"
              />
            </svg>
             by
             
            <a
              href="https://carloscuesta.me"
            >
              Carlos Cuesta
            </a>
          </h3>
        </div>
        <div
          className="col-sm-6 footer-nav"
        >
          <nav>
            <a
              href="/about"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              About
            </a>
            <a
              href="/contributors"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Contributors
            </a>
            <a
              href="https://github.com/carloscuesta/gitmoji"
            >
              GitHub
            </a>
          </nav>
        </div>
      </div>
    </div>
  </footer>,
]
`;
